File: owner-app-flutter/lib/features/messages/models/message.dart
class PetMessage {
  const PetMessage({
    required this.messageId,
    required this.messageThreadId,
    required this.senderUserId,
    required this.senderName,
    required this.body,
    this.videoSubmissionId,
    this.attachmentUrl,
    this.attachmentName,
    this.attachmentType,
    this.readAt,
    required this.createdDate,
  });

  final int messageId;
  final int messageThreadId;
  final int senderUserId;
  final String senderName;
  final String body;
  final int? videoSubmissionId;
  final String? attachmentUrl;
  final String? attachmentName;
  final String? attachmentType;
  final DateTime? readAt;
  final DateTime createdDate;

  factory PetMessage.fromJson(Map<String, dynamic> json) {
    return PetMessage(
      messageId: json['messageId'] as int,
      messageThreadId: json['messageThreadId'] as int,
      senderUserId: json['senderUserId'] as int,
      senderName: json['senderName'] as String,
      body: json['body'] as String,
      videoSubmissionId: json['videoSubmissionId'] as int?,
      attachmentUrl: json['attachmentUrl'] as String?,
      attachmentName: json['attachmentName'] as String?,
      attachmentType: json['attachmentType'] as String?,
      readAt: json['readAt'] != null ? DateTime.parse(json['readAt'] as String) : null,
      createdDate: DateTime.parse(json['createdDate'] as String),
    );
  }
}

File: physio-portal-vue3/src/types/message.ts
export interface Message {
  messageId: number
  messageThreadId: number
  senderUserId: number
  senderName: string
  body: string
  videoSubmissionId: number | null
  videoTitle?: string | null
  attachmentUrl?: string | null
  attachmentName?: string | null
  attachmentType?: string | null
  readAt: string | null
  createdDate: string
}

export interface MessageThread {
  messageThreadId: number
  petId: number
  petName: string
  ownerId: number
  ownerName: string
  physioId: number
  physioName: string
  lastMessagePreview: string | null
  lastMessageAt: string | null
  unreadCount: number
}

export interface SendMessageRequest {
  body: string
  videoSubmissionId?: number
  attachmentUrl?: string
  attachmentName?: string
  attachmentType?: string
}
